security/advancedtls/advancedtls_test.go

```diff
@@ -77,34 +77,42 @@ func TestClientServerHandshake(t *testing.T) {
 		clientRoot                 *x509.CertPool
 		clientGetRoot              func(params *GetRootCAsParams) (*GetRootCAsResults, error)
 		clientVerifyFunc           CustomVerificationFunc
+		clientVType                VerificationAuthType
 		clientExpectCreateError    bool
 		clientExpectHandshakeError bool
 		serverMutualTLS            bool
 		serverCert                 []tls.Certificate
 		serverGetCert              func(*tls.ClientHelloInfo) (*tls.Certificate, error)
 		serverRoot                 *x509.CertPool
 		serverGetRoot              func(params *GetRootCAsParams) (*GetRootCAsResults, error)
+		serverVerifyFunc           CustomVerificationFunc
+		serverVType                VerificationAuthType
 		serverExpectError          bool
 	}{
 		// Client: nil setting
 		// Server: only set serverCert with mutual TLS off
 		// Expected Behavior: server side failure
-		// Reason: if either clientCert or clientGetClientCert is not set and
-		// verifyFunc is not set, we will fail directly
+		// Reason: if |clientRoot|, |clientGetRoot| and |verifyFunc| is not set, client side
+		// doesn't provide any verification mechanism. We don't allow this even setting
+		// |vType| to SkipVerification. Clients should at least provide their own
+		// verification logic.
 		{
 			"Client_no_trust_cert_Server_peer_cert",
 			nil,
```

Comment:
You can use the member names of the struct to simplify the test cases :) For example, this use case can be simplified as

```
{
    desc: "Client_no_trust_cert_Server_peer_cert",
    clientVType: SkipVerification,
    serverCert: []tls.Certificate{serverPeerCert},
    serverVType: CertAndHostVerification,
    serverExpectError: true,
}

Feel free to create a separate PR for this change since it's not very related to this one.

Reply:
Thanks for the suggestions! I didn't notice this before because I have my IDE doing the field name completion for me:-)
I will make another PR for this later.

Reply:
SG, thanks. Feel free to send it to me for review.